### DISCUSSION TITLE
你如何设置环境变量？

### DISCUSSION THREAD

**Discussion by xXAvoraXx** [umijs/umi#11568] [Q&A] [answered]
您好，虽然我在根目录中创建了“.env”文件，但我无法达到“process.env.TEST”值，您能帮忙吗？

// .env
TEST="TEST"

// some component
const test = process.env.TEST;
console.log(test) 
"undefined"

**Comment by fz6m** [comment]
env 文件里写的变量，只是在 node 里能用（比如在 `.umirc.ts` 配置文件里），但是在项目里还用不了。

其中 `UMI_APP_` 开头的会被自动加载到项目里，其他的变量需要通过 [`define`](https://umijs.org/docs/api/config#define) 声明才能注入到项目里。

**Reply by xXAvoraXx** [reply]
我可以在哪个文件中进行 UMI_APP_ 定义？

**Reply by fz6m** [reply, marked as answer]
`.env` 和 `.env.development` / `.env.production` 都可以。

**Reply by xXAvoraXx** [reply]
感谢您的帮助。 我会尽快尝试。 我们可以将应用程序中的“UMI_APP_”值作为“process.env.UMI_APP_”访问吗？

**Reply by fz6m** [reply]
如果你在 `.env` 定义了 `UMI_APP_XXX` ，则无需配置 `define` 也可以直接在应用中使用 `process.env.UMI_APP_XXX` ，否则，还需在 `define` 中配置，请参考文档 [代码示例](https://umijs.org/docs/api/config#define) 。

**Reply by xXAvoraXx** [reply]
感谢您的信息。
